internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/fastclient/AvroStoreClientEndToEndTest.java

```diff
@@ -215,6 +218,111 @@ protected void runTest(
     }
   }
 
+  private void runTestWithClientException(
+      ClientConfig.ClientConfigBuilder clientConfigBuilder,
+      boolean batchGet,
+      Consumer<MetricsRepository> statsValidation,
+      Optional<AvroGenericStoreClient> vsonThinClient) throws Exception {
+    MetricsRepository metricsRepositoryForGenericClient = new MetricsRepository();
+    AvroGenericStoreClient<String, GenericRecord> genericFastClient =
+        getGenericFastClient(clientConfigBuilder, metricsRepositoryForGenericClient, true);
+
+    AvroGenericStoreClient<String, Object> genericFastVsonClient = null;
+    // Construct a Vson store client
+    genericFastVsonClient =
+        getGenericFastVsonClient(clientConfigBuilder.clone(), new MetricsRepository(), vsonThinClient, true);
+    try {
+      if (batchGet) {
+        Set<String> keys = new HashSet<>();
+        for (int i = 0; i < recordCnt; ++i) {
+          String key = keyPrefix + i;
+          keys.add(key);
+        }
+
+        try {
+          genericFastClient.batchGet(keys).get();
+          fail();
+        } catch (VeniceClientException | ExecutionException e) {
+          assertTrue(e.getMessage().endsWith("metadata is not ready, attempting to re-initialize"));
+        }
+        // Test Vson client
+        try {
+          genericFastVsonClient.batchGet(keys).get();
+          fail();
+        } catch (VeniceClientException | ExecutionException e) {
+          assertTrue(e.getMessage().endsWith("metadata is not ready, attempting to re-initialize"));
+        }
+      } else {
+        for (int i = 0; i < recordCnt; ++i) {
+          String key = keyPrefix + i;
+
+          try {
+            genericFastClient.get(key).get();
+            fail();
```

Comment:
Can we reuse the `runTest` and add `expectedExceptions` to `testFastClientWithoutServers`'s Test annotation rather than adding a whole new function and checking the exception inline?

Reply:
Made the refactor to reuse the method

Reply:
As discussed offline, please make changes like [this](https://github.com/linkedin/venice/blob/main/clients/venice-push-job/src/test/java/com/linkedin/venice/hadoop/TestValidateSchemaAndBuildDictMapperOutputReader.java#L19). You don't have to handle whether you are getting the exception separately and let `expectedExceptions` handle it.